Library/Homebrew/cli_parser.rb

```diff
@@ -0,0 +1,280 @@
+module HomebrewCli
+  class Parser
+    attr_accessor :ignore_invalid_options
+    attr_reader :unnamed
+
+    def initialize(&block)
+      @specs = {}
+      @long = {}
+      @short = {}
+      @order = []
+      @constraints = []
+
+      instance_eval(&block)
+    end
+
+    def option(name, desc = "", opts = {})
```

Comment:
Generate `desc` from the name, by default at least. Consider using named parameters over an options hash.

Reply:
Addressed.